src/controller/CHIPDeviceController.cpp

```diff
@@ -886,7 +886,7 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
     VerifyOrExit(exchangeCtxt != nullptr, err = CHIP_ERROR_INTERNAL);
 
     err = device->GetPairing().Pair(params.GetPeerAddress(), params.GetSetupPINCode(), keyID,
-                                    Optional<ReliableMessageProtocolConfig>::Value(mMRPConfig), exchangeCtxt, this);
+                                    Optional<ReliableMessageProtocolConfig>::Value(GetLocalMRPConfig()), exchangeCtxt, this);
```

Comment:
I thought this was supposed to take the peer's MRP settings, e.g. as discovered through DNS-SD.  No?

Reply:
Yeah, I guess it should be possible to pass this value by `RendezvousParams` which would be filled by `SetupCodePairer`. I think the controller part should be additionally revised as we probably don't properly handle the case where the controller is SED (the controller should probably use MAX(local MRP config, peer MRP config)) for MRP. I can file an issue for that.

Reply:
It already exists: https://github.com/project-chip/connectedhomeip/issues/13941

Reply:
I think if the peer doesn't advertise MRP parameters then we should take our local node MRP config. For example, if the initiator is SED with polling interval `10s` it should do the retransmission only after `10s + C` because even if everything is OK with the network, the initiator may not be able to receive the ACK earlier than after `10s` due to its polling interval. And if the peer advertises MRP parameters, the initiator should use these parameters increased by the local polling interval (if any). Am I right?

Reply:
> I think if the peer doesn't advertise MRP parameters then we should take our local node MRP config. For example, if the initiator is SED with polling interval 10s it should do the retransmission only after 10s + C because even if everything is OK with the network, the initiator may not be able to receive the ACK earlier than after 10s due to its polling interval. And if the peer advertises MRP parameters, the initiator should use these parameters increased by the local polling interval (if any). Am I right?

Sorry for the slow reply.  But this is a very surprising interpretation for me.  I think that for sleepy nodes to successfully communicate, they need to be awake at the times the peer needs them to be awake, and this implies either not sleeping while MRP is underway, or at least waking on some interval that is compatible with the peer's advertised MRP needs.

This implies to me that a node would never take into account their own MRP settings when considering their peer's MRP settings.

Am I missing something?

@turon , @tcarmelveilleux ?

Reply:
The spec only mentions changing the polling interval (from slow-polling to fast-polling) when there is an active exchange. There's nothing about disabling SED mode while MRP is underway. To achieve low power consumption SED must wake up for a very short time, just enough to ask a parent router if there are any incoming messages buffered, so it would be difficult for a peer node behind a few routers to know when to send the message to hit the wake-up period. If there is a miss, then we can't deliver the message (and get a response) for the next polling interval. That's why the spec mentions that the advertised MRP parameters for a SED should be at least the fast/slow-polling interval.

But you're right that the spec doesn't seem to be concerned about SED polling intervals of a sender when determining the retransmission timeouts. However, when I think about it, it seems to me that a SED sender should take into account its fast-polling interval. Just like the outgoing message can be delayed by the recipient's polling interval, so can the response be delayed by the sender's polling interval.